Model/Sync/Reset/Main.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync\Reset;
+
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\App\ResourceConnection;
+use Yotpo\Core\Model\Config;
+
+class Main
+{
+    const DELETE_LIMIT = 10000;
+    const UPDATE_LIMIT = 2000;
+
+    /**
+     * @var ResourceConnection
+     */
+    protected $resourceConnection;
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var TypeListInterface
+     */
+    protected $cacheTypeList;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        Config $config,
+        TypeListInterface $cacheTypeList
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->config = $config;
+        $this->cacheTypeList = $cacheTypeList;
+    }
+
+    /**
+     * @param int $storeId
+     * @param boolean $skipSyncTables
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function resetSync($storeId, $skipSyncTables = false)
+    {
+        $this->setResetInProgressConfig($storeId, '1');
+        $this->deleteRunningCronSchedules();
+        if (!$skipSyncTables) {
+            $this->deleteAllFromTables($storeId);
+        }
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getTableResourceNames()
+    {
+        return [];
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getCronJobCodes()
+    {
+        return [];
+    }
+
+    /**
+     * @return string
+     */
+    protected function getYotpoEntityName()
+    {
+        return '';
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function deleteAllFromTables($storeId)
+    {
+        $tableResourceNames = $this->getTableResourceNames();
+        foreach ($tableResourceNames as $tableResourceName) {
+            $this->deleteAllFromTable($storeId, $tableResourceName);
+        }
+    }
+
+    /**
+     * @param int $storeId
+     * @param string $tableResourceName
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function deleteAllFromTable($storeId, $tableResourceName)
+    {
+        $tableName = $this->resourceConnection->getTableName($tableResourceName);
+        $totalCount = $this->getTotalCount($tableName, $storeId);
```

Comment:
Do we have to count here the whole table? This can cause performance issues.
No other way you can think of further than counting the number of records in the table?

Reply:
Without taking count, we won't be able to apply the batch delete logic here.
Otherwise we would need to delete by batch and run a select query in each iteration until the entire table becomes empty